pkg/kubectl/cmd/util/printing.go

```diff
@@ -83,8 +86,11 @@ func OutputVersion(cmd *cobra.Command, defaultVersion string) string {
 func PrinterForCommand(cmd *cobra.Command) (kubectl.ResourcePrinter, bool, error) {
 	outputFormat := GetFlagString(cmd, "output")
 	templateFile := GetFlagString(cmd, "template")
-	if len(outputFormat) == 0 && len(templateFile) != 0 {
-		outputFormat = "template"
+	if len(templateFile) != 0 {
+		glog.Warningf("%s is DEPRECATED and will be removed in a future version.", "-t")
```

Comment:
Is there way for you to tell whether they've typed `-t` versus `--template`.  If I'm reading the issue right, the intent is to deprecate `-t`, but not `--template`.  I'm concerned about dumping this message too aggressively.

One alternative is to actually split `--template` into two flags and use `MarkDeprecated` for the `-t` flag.  If you do that, pflags will take care of the deprecation notification.